app/src/main/java/eu/kanade/tachiyomi/source/online/english/Mangadex.kt

```diff
@@ -324,6 +326,101 @@ open class Mangadex(override val lang: String, private val internalLang: String,
+        return getFollowsClient(filters).newCall(followsListRequest(page, filters))
+                .asObservable()
+                .map { response ->
+                    followsParse(response)
+                }.flatMap { parsedResponse ->
+                    // TODO I have no idea if this will work properly or if there's a better way
+                    val currentObservable = Observable.just(parsedResponse.first)
+                    if (parsedResponse.second)
+                        currentObservable.mergeWith(fetchFollows(page + 1, filters))
+                    else
+                        currentObservable
+                }
+    }
+
+    fun followsParse(response: Response): Pair<List<Pair<SManga, Int>>, Boolean> { // TODO exit generic hell
+        val document = response.asJsoup()
+
+        val follows = document.select(followSelector()).map { element ->
+            followFromElement(element)
+        }
+
+        val hasNextPage = followsNextPageSelector()?.let { selector ->
+            document.select(selector).first()
+        } != null
+
+        return Pair(follows, hasNextPage)
+    }
+
+    private fun getFollowsClient(filters: FilterList): OkHttpClient { // TODO reuse/rename `getSearchClient()`?
+        filters.forEach { filter ->
+            when (filter) {
+                is R18 -> {
+                    return when (filter.state) {
+                        1 -> clientBuilder(ALL)
+                        2 -> clientBuilder(ONLY_R18)
+                        3 -> clientBuilder(NO_R18)
+                        else -> clientBuilder()
+                    }
+                }
+            }
+        }
+        return clientBuilder()
+    }
+
+    fun followsListRequest(page: Int, filters: FilterList): Request {
+
+        // Format `/follows/manga/$status[/$sort[/$page]]`
+        val url = HttpUrl.parse("$baseUrl/follows/manga/0")!!.newBuilder() // Gets regardless of follow status. TODO add filter?
+                .addQueryParameter("p", page.toString())
+
+        filters.forEach { filter ->
+            when (filter) {
+                is TextField -> url.addQueryParameter(filter.key, filter.state)
+                is SortFilter -> {
+                    if (filter.state != null) {
+                        if (filter.state!!.ascending) {
+                            url.addQueryParameter("s", sortables[filter.state!!.index].second.toString())
+                        } else {
+                            url.addQueryParameter("s", sortables[filter.state!!.index].third.toString())
+                        }
+                    }
+                }
+            }
+        }
+
+
+        return GET(url.toString(), headersBuilder().build())
+    }
+
+    private fun followSelector() = "div.manga-entry"
+
+    private fun followFromElement(element: Element): Pair<SManga, Int> { // TODO can follow status be added to `SManga`?
```

Comment:
See no issue adding follow status.  Will need to change it multiple places though and add to database